src/Whim/Layout/FreeLayoutEngine.cs

```diff
@@ -0,0 +1,145 @@
+﻿using System.Linq;
+using DotNext.Collections.Generic;
+
+namespace Whim;
+
+/// <summary>
+/// Layout engine that lays out all windows as free-floating.
+/// </summary>
+public class FreeLayoutEngine : ILayoutEngine
+{
+	private readonly IContext _context;
+	private readonly ImmutableDictionary<IWindow, IRectangle<double>> _dict;
+
+	/// <inheritdoc/>
+	public string Name { get; init; } = "Free";
+
+	/// <inheritdoc/>
+	public int Count => _dict.Count;
+
+	public LayoutEngineIdentity Identity { get; }
+
+	/// <summary>
+	/// Creates a new instance of the <see cref="FreeLayoutEngine"/> class.
+	/// </summary>
+	/// <param name="context">The identity of the layout engine.</param>
+	/// <param name="identity">The context of the layout engine.</param>
+	public FreeLayoutEngine(IContext context, LayoutEngineIdentity identity)
+	{
+		Identity = identity;
+		_context = context;
+		_dict = ImmutableDictionary<IWindow, IRectangle<double>>.Empty;
+	}
+
+	private FreeLayoutEngine(FreeLayoutEngine layoutEngine, ImmutableDictionary<IWindow, IRectangle<double>> dict)
+	{
+		Name = layoutEngine.Name;
+		Identity = layoutEngine.Identity;
+		_context = layoutEngine._context;
+		_dict = dict;
+	}
+
+	/// <inheritdoc/>
+	public ILayoutEngine AddWindow(IWindow window)
+	{
+		Logger.Debug($"Adding window {window} to layout engine {Name}");
+
+		if (_dict.ContainsKey(window))
+		{
+			Logger.Debug($"Window {window} already exists in layout engine {Name}");
```

Comment:
Depending on the file, sometimes it is a "Debug" and sometimes a "Error".
Which one should be used?
![image](https://github.com/dalyIsaac/Whim/assets/2392459/f079f601-ab0d-46c5-8c65-2ca612f15aec)

Reply:
Whoops, `Debug` is fine.